redshift_sqlalchemy/dialect.py

```diff
@@ -19,6 +23,53 @@ class RedshiftImpl(postgresql.PostgresqlImpl):
         __dialect__ = 'redshift'
 
 
+## Constants
+
+COMPOUND_SORTKEY = 'compound'
+INTERLEAVED_SORTKEY = 'interleaved'
+
+
+## Regular expressions
+
+
+# For parsing adsrc, e.g.: "identity"(445178, 0, '1,1'::text)
+IDENTITY_RE = re.compile(r"""
+    "identity" \(
+      (?P<current>-?\d+)
+      ,\s
+      (?P<base>-?\d+)
+      ,\s
+      '(?P<seed>-?\d+),(?P<step>-?\d+)'
+      .*
+    \)
+""", re.VERBOSE)
+
+# For constraints, e.g.:
+#   FOREIGN KEY(col1) REFERENCES othertable (col2)
+# See https://docs.aws.amazon.com/redshift/latest/dg/r_names.html
+# for a definition of valid SQL identifiers.
+FK_RE = re.compile(r"""
+  ^FOREIGN KEY \s* \(    # FOREIGN KEY, arbitrary whitespace, literal '('
+    (?P<columns>         # Start a group to capture the referring columns
+      (?:                # Start a non-capturing group
+        \s*              # Arbitrary whitespace
+        [_a-zA-Z][\w$]*  # SQL identifier
```

Comment:
According to http://docs.aws.amazon.com/redshift/latest/dg/r_names.html:

Standard SQL identifiers adhere to a set of rules and must:

Contain only ASCII letters, digits, underscore characters (_), or dollar signs ($).

Begin with an alphabetic character or underscore character. Subsequent characters may include letters, digits, underscores, or dollar signs.

Be between 1 and 127 characters in length, not including quotes for delimited identifiers.

Contain no quotation marks and no spaces.

Not be a reserved SQL key word.